code/utils/scene_slicer.py

```diff
@@ -1,74 +1,69 @@
+from __future__ import division
 import nibabel as nib
 import csv
 import numpy as np
 import json
+import math
 
-IS_DAY = 0
-IS_INT = 1
+INTEGER_LABELS = {'day-night': {'DAY': 0,
+                                'NIGHT': 1,
+                                'DAWN': 2},
+                  'int-ext': {'INT': 0,
+                              'EXT': 1}}
 
 DAY_NIGHT_IND = 0
 INT_EXT_IND = 1
 
+TUNING_SECONDS_OFFSET = 17
+
 # with open('../../data/data_path.json', 'r') as fh:
 #     data_paths = json.load(fh)
 # path_to_subject_image = "../../" + data_paths['bold_dico_7Tad2grpbold7Tad']['sub1']['runs'][0]['path']
 
 
 class SceneSlicer:
-    def __init__(
-            self,
-            path_to_subject_image,
-            path_to_scene_csv="../../ds113_study_description/stimulus/task001/annotations/scenes.csv"
-    ):
+    def __init__(self, path_to_subject_image, path_to_scene_csv):
         self.path_to_scene_csv = path_to_scene_csv
         self.image = nib.load(path_to_subject_image)
         self.scene_slices = []
-        self.segment_duration = [902, 882, 876, 976, 924, 878, 1086, 673.4]
         self.scene_desc = {}
-        self.scene_keys = []
 
-    def generate_scene_desc_dict(self):
+    def generate_scene_desc_dict_(self):
         with open(self.path_to_scene_csv, 'rt') as csvfile:
             reader = csv.DictReader(
                 csvfile,
                 fieldnames=['seconds', 'scene', 'day-night', 'int-ext'])
             for row in reader:
                 scene_time = int(float(row['seconds']))
-                self.scene_desc[scene_time] = (row['day-night'] == "DAY",
-                                               row['int-ext'] == "INT")
-        self.scene_keys = list(self.scene_desc.keys())
-        self.scene_keys.sort()
+                self.scene_desc[scene_time] = (
+                    INTEGER_LABELS['day-night'][row['day-night']],
+                    INTEGER_LABELS['int-ext'][row['int-ext']])
+
+    def generate_scene_slices_(self):
+        num_offset_slices = int(math.floor(TUNING_SECONDS_OFFSET / 2))
+        day_night = num_offset_slices * [None]
+        int_ext = num_offset_slices * [None]
+        current_scene_start_time = TUNING_SECONDS_OFFSET
+        for i in range(num_offset_slices, self.image.shape[-1]):
+            if i * 2 + 1 in self.scene_desc:
+                current_scene_start_time = i * 2 + 1
+            elif i * 2 + 2 in self.scene_desc:
+                current_scene_start_time = i * 2 + 2
+            day_night.append(self.scene_desc[current_scene_start_time][0])
+            int_ext.append(self.scene_desc[current_scene_start_time][1])
```

Comment:
it would be better to actually save self.scene_desc[scene_start_time] instead of only saving current_scene_start_time, but it's not a big deal. i'll approve anyway